physical/postgresql/postgresql.go

```diff
@@ -96,10 +147,28 @@ func NewPostgreSQLBackend(conf map[string]string, logger log.Logger) (physical.B
 		get_query:    "SELECT value FROM " + quoted_table + " WHERE path = $1 AND key = $2",
 		delete_query: "DELETE FROM " + quoted_table + " WHERE path = $1 AND key = $2",
 		list_query: "SELECT key FROM " + quoted_table + " WHERE path = $1" +
-			"UNION SELECT DISTINCT substring(substr(path, length($1)+1) from '^.*?/') FROM " +
-			quoted_table + " WHERE parent_path LIKE $1 || '%'",
+			" UNION SELECT DISTINCT substring(substr(path, length($1)+1) from '^.*?/') FROM " + quoted_table +
+			" WHERE parent_path LIKE $1 || '%'",
+		haGetLockIdentityQuery:
+		//only read non expired data
+		" SELECT ha_identity FROM vault_ha_store WHERE NOW() <= valid_until AND ha_key = $1 ",
+		haGetLockValueQuery:
+		//only read non expired data
+		" SELECT ha_value FROM vault_ha_store WHERE NOW() <= valid_until AND ha_key = $1 ",
+		haUpsertLockIdentityExec:
+		// $1=identity $2=ha_key $3=ha_value $4=TTL in seconds
+		//update either steal expired lock OR update expiry for lock owned by me
+		" INSERT INTO vault_ha_store (ha_identity, ha_key, ha_value, valid_until) VALUES ($1, $2, $3, NOW() + $4 * INTERVAL '1 seconds'  ) " +
+			" ON CONFLICT (ha_key) DO " +
+			" UPDATE SET (ha_identity, ha_key, ha_value, valid_until) = ($1, $2, $3, NOW() + $4 * INTERVAL '1 seconds') " +
+			" WHERE vault_ha_store.valid_until < NOW() AND vault_ha_store.ha_key = $2 OR " +
```

Comment:
Please add parens here rather than relying on operator precedence.

Reply:
Agree, makes it more readable, less prone to bugs.